Update tree-sitter-java and add Java textobjects

Ordoviz opened the pull request:
tree-sitter-java [gained support](https://github.com/tree-sitter/tree-sitter-java/compare/bd6186c24d5eb13b4623efac9d944dcc095c0dad...09d650def6cdf7f479f4b78f595e9ef5b58ce31e) for Java """text blocks""", switch expressions, and records, so I figured it was time to update it.

the-mikedavis reviewed (commented):
It looks like `yield` was also added [tree-sitter/tree-sitter-java@`bd6186c24d5eb13b4623efac9d944dcc095c0dad...09d650def6cdf7f479f4b78f595e9ef5b58ce31e`#diff-919ac210ac](https://github.com/tree-sitter/tree-sitter-java/compare/bd6186c24d5eb13b4623efac9d944dcc095c0dad...09d650def6cdf7f479f4b78f595e9ef5b58ce31e#diff-919ac210accac9ecc55a76d10a7590e3d85ca3f0e165b52d30f08faee486d0cbR487-R491) which could be added to the `keyword` highlights

Ordoviz commented:
> It looks like `yield` was also added

Good catch. I added the  `permits` and `yield` keywords. Helix now highlights all keywords listed on [Wikipedia](https://en.wikipedia.org/wiki/List_of_Java_keywords).
